src/xtopcom/xstatistic/src/xclass_type_converter.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2017-2020 Telos Foundation & contributors
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "xstatistic/xclass_type_converter.h"
+#include "xrpc/xmessage_ids.h"
+#include "xstate_sync/xmessage_ids.h"
+#include "xsyncbase/xmessage_ids.h"
+#include "xtxpool_v2/xmessage_ids.h"
+#include "xunit_service/xmessage_ids.h"
+#include "xvm/manager/xmessage_ids.h"
+
+NS_BEG2(top, xstatistic)
+
+#if defined(CACHE_SIZE_STATISTIC) || defined(CACHE_SIZE_STATISTIC_MORE_DETAIL)
+enum_statistic_class_type message_id_to_class_type(uint32_t msgid) {
+    uint32_t message_category = (msgid & 0xFFFF0000) >> 16;
+
+#ifndef CACHE_SIZE_STATISTIC_MORE_DETAIL
+    switch (message_category) {
+        case (uint32_t)xmessage_category_consensus :
+        case (uint32_t)xmessage_category_timer :
+        case (uint32_t)xmessage_category_relay :
+            return enum_statistic_msg_cons;
+        case (uint32_t)xmessage_category_txpool :
+            return enum_statistic_msg_txpool;
+        case (uint32_t)xmessage_category_rpc :
+            return enum_statistic_msg_rpc;
+        case (uint32_t)xmessage_category_sync :
+            return enum_statistic_msg_sync;
+        case (uint32_t)xmessage_block_broadcast :
+            return enum_statistic_msg_block_broadcast;
+        case (uint32_t)xmessage_category_state_sync :
+            return enum_statistic_msg_state;
+        default :
+            xerror("msgid:%u", msgid);
+            return enum_statistic_undetermined;     
+    }
+#else
+    // value of msgid of bft message is added pdu_type, so that here can not directly use xBFT_msg, xTimer_msg and xrelay_BFT_msg.
+    switch (message_category) {
+        case (uint32_t)xmessage_category_consensus :
+            return enum_statistic_msg_bft;
+        case (uint32_t)xmessage_category_timer :
+            return enum_statistic_msg_timer;
+        case (uint32_t)xmessage_category_relay :
+            return enum_statistic_msg_relay_bft;
+        default :
+            break;
+    }
+
+    switch (msgid) {
+        case (uint32_t)common::xmessage_id_t::invalid:
+            return enum_statistic_undetermined;
+        case (uint32_t)xrpc::rpc_msg_request :

```

Comment:
since xrpc::rpc_msg_request is defined in xrpc module, xstatistic relies on xrpc.  but will xrpc rely on xstatistic due to using xstatistic functionalities?

Reply:
fixed.